game-core/src/main/java/games/strategy/engine/framework/GameRunner.java

```diff
@@ -81,30 +81,27 @@ private GameRunner() {}
    */
   public static void start() {
     SwingUtilities.invokeLater(() -> {
-      final JFrame frame = JFrameBuilder.builder()
-          .title("TripleA")
-          .build();
-      setupPanelModel = new SetupPanelModel(gameSelectorModel, frame);
-      mainFrame = newMainFrame(frame);
+      newMainFrame();
+      setupPanelModel = new SetupPanelModel(gameSelectorModel, mainFrame);
+      mainFrame.add(new MainPanelBuilder().buildMainPanel(setupPanelModel, gameSelectorModel));
+      mainFrame.pack();
       setupPanelModel.showSelectType();
       new Thread(GameRunner::showMainFrame).start();
     });
 
     UpdateChecks.launch();
   }
 
-  private static JFrame newMainFrame(final JFrame frame) {
-    LookAndFeelSwingFrameListener.register(frame);
+  public static void newMainFrame() {
+    mainFrame = JFrameBuilder.builder()
+        .title("TripleA")
+        .build();
+    LookAndFeelSwingFrameListener.register(mainFrame);
 
-    frame.add(new MainPanelBuilder().buildMainPanel(setupPanelModel, gameSelectorModel));
-    frame.pack();
+    mainFrame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
```

Comment:
(1) it would be great to see the builder enhanced so that this kind of code could be inline with the builder and not be its own call.

(2) are we really sure we want exit on close? For example when connecting to the lobby, the main frame is closed, would the game then exit in that case?

Reply:
> are we really sure we want exit on close? For example when connecting to the lobby, the main frame is closed, would the game then exit in that case?

At least currently we are. When connecting to a game in the lobby, a new JVM is created, so when calling `System.exit(0)` on either process the other one is obviously not affected.
This is ugly, but almost all of the code is expecting the engine to have only a single game instance per process.

Reply:
To be sure, I'm wondering about the transition from the loading screens to the lobby screen. IIRC connecting to the lobby is a new frame and not a new instance. I do agree we have new processes for the game instances, but that is only after clicking through to get to a staging screen. Let me know if I'm wrong about the lobby window being same process.

Reply:
Coming back at this I'm not sure if I can give a good wanser on this one. I mean this is pre-existing code, so I guess it works?